### PR TITLE
When the tab receives focus, immediately move focus to the active control

### LINKED ISSUE #3609: Clicking the current tab to raise the window should focus the text area
# Environment

```none
Windows build number: 10.0.18363.0
Windows Terminal version: 0.6.2951.0

Any other software? Nope
```

# Steps to reproduce

1. Start Terminal
1. Focus another window
1. Re-focus Terminal by clicking on the active tab
1. Enter text using your keyboard

# Expected behavior

Raising the Terminal window should focus the text area no matter how you do it.

Works as expected if you click any other tab than the currently active.

# Actual behavior

Text input is ignored even though the Terminal window has focus. Text input is restored when clicking the text area or focusing another tab.

### LINKED ISSUE #5750: ESC from the Tab ContextMenu should return focus to the Terminal
### Repro steps

1. Right click a tab
2. Hit <kbd>esc</kbd>

#### Expected
3. Terminal gets focus again

#### Actual
3. Focus lands on the tab itself
image

regressed in #3789

Also, fixing this might just magically fix #3609.

### PR DESCRIPTION
**Pull request by zadjii-msft**
I'm mostly just putting the PR out there for comments. I'm worried that this might have horrifying unintentional side-effects for accessibility, which I'm not loving. 

This PR makes the `Tab` immediately toss focus to the active control, whenever the `Tab` receives focus, for any reason. 
* Left-click on the tab to focus the window? Focus the control. #3609
* Close the color picker? Focus the control.
* Hit esc to dismiss the menuflyout from the New Tab menu, the tab context menu? Focus the control. #5750
* Hit esc to dismiss the tab rename? Focus the control. 

@carlos-zamora will this horribly break accessibility?

My other theory here is to pre-emptively prevent the tab from receiving focus. That would probably involve
* Focusing the control when a tab is left clicked
* Manually passing focus to the control when any flyout is closed
* Manually passing focus to the control when the color picker is closed
* Manually passing focus to the control when the tab renamer is dismissed
* Manually passing focus to the control when the command palette is closed
* Manually passing focus to the control when \<any future UI> is closed

So that might not scale as well, but that might be the more _correct_ solution. 

* [x] I work here
* [ ] This is UI so it doesn't have tests
* [x] Closes #3609
* [x] Closes #5750

### CODE REVIEW

**Comment by miniksa** [comment]
I don't know about this. I recall there being some UI paradigm where you can use the tab key to move around between UI elements. And if that element is a tab row, you can then use the arrow keys to move a highlight between the tabs and press spacebar to then switch to the highlighted one. And if you're shirking the focus off the tab row anytime it hits... that's probably broken. I don't know if that's a part of the TabView control that we get from MUX, but I'm doing that right now in Microsoft Edge and it works that way...

**Comment by zadjii-msft** [comment]
Yea, that was basically exactly my feeling with this PR. It seemed like "it worked", but not the _right_ way. 

Unless anyone else has any enlightened ideas about how to do this in a more clever way, I think I'm going to change this to the more manual "this control knows it's been closed, so it'll ask for focus to go back to the Terminal". It'll be more manual work to do for each and every UI element, but seems like the only good solution.

**Comment by carlos-zamora** [comment]
Downloaded and did some testing:
- **Narrator**:
   - user can use the <kbd>Narrator+Left</kbd> and <kbd>Narrator+Right</kbd> commands to navigate to other UI elements. So the user is still able to access the tabs (aside from using our custom keybindings to invoke them). ([reference](https://support.microsoft.com/en-us/help/22806/windows-10-narrator-keyboard-commands-touch-gestures))
- **NVDA**: 
   - The good news is that NVDA already separates UI navigation from where the focus actually is (there's an NVDA keybinding to specifically navigate to the focused object). So, this basically doesn't affect it.
   - (unrelated to this PR) The bad news is, NVDA's UI navigation model exposes so many ugly details in our UI (i.e.: the cutout is something you can navigate to as an "unknown" object). This makes finding the tabs a pain (Tab Row Control --> list --> <the actual tabs>), and I wasn't able to find the dropdown. The workaround here though is to use the dropdown keybinding.
   - All of these problems already existed before though. So I'll file an issue to track that specifically since it's unrelated to this PR.

TL;DR: this did _not_ horribly break accessibility

**Comment by mle-ii** [comment]
Is this just waiting on the reviewer to approve?  Besides the merge conflicts, or is there some other blocker here?

**Comment by zadjii-msft** [comment]
I've kinda just left this stagnate while we were building some other UI elements, because this probably needs to be solved in a way that's more cohesive across various different UI elements. Most of that work is being tracked in #6680. I want to make sure that however we actually do end up fixing this is the right solution, not just a hack, and this fix seems like a hack ATM.

**Comment by zadjii-msft** [comment]
Closing in favor of the much more targeted strike in #10048
